File: java/knapsack/src/main/java/Knapsack.java
import java.util.List;
import java.util.stream.IntStream;

public class Knapsack {
  public int maximumValue(int capacity, List<Item> items) {
    return IntStream.range(0, 1 << items.size())
        .map(
            mask -> {
              int weightSum =
                  IntStream.range(0, items.size())
                      .filter(i -> ((mask >> i) & 1) == 1)
                      .map(i -> items.get(i).weight)
                      .sum();
              int valueSum =
                  IntStream.range(0, items.size())
                      .filter(i -> ((mask >> i) & 1) == 1)
                      .map(i -> items.get(i).value)
                      .sum();

              return (weightSum <= capacity) ? valueSum : -1;
            })
        .max()
        .getAsInt();
  }
}

File: python/knapsack/knapsack.py
def maximum_value(maximum_weight, items):
    result = -1
    for code in range(1 << len(items)):
        used = decode(len(items), code)

        weight_sum = sum(items[i]['weight']
                         for i in range(len(items)) if used[i])
        if weight_sum <= maximum_weight:
            value_sum = sum(items[i]['value']
                            for i in range(len(items)) if used[i])
            result = max(result, value_sum)

    return result


def decode(size, code):
    used = []
    for _ in range(size):
        used.append(code & 1 != 0)
        code >>= 1
    return used
